Neos.Flow/Classes/Security/Authentication/AuthenticationManagerInterface.php

```diff
@@ -26,13 +28,17 @@
      * Note: The order of the tokens in the array is important, as the tokens will be authenticated in the given order.
      *
      * @return array<TokenInterface> An array of tokens this manager is responsible for
+     * @deprecated Use the TokenAndProviderFactory
```

Comment:
I'm not so sure about deprecating this API (and the `getProviders()`) - IMO the AuthenticationManager is the main entry point for all Authentication related API calls and it's not quite clear why there is a "TokenAndProviderFactory" (and then why one Factory for both?) that one needs to inject and use. This is a good and useful implementation detail that helps with maintenance, but we should think twice before widening the API surface in such a way.

```suggestion
```

Reply:
is there a reason someone would want that info though? I see it more as an internal thing anyway and actually wanted to reduce the api surface by making the factory non api for now.

Reply:
I think that's fine…